### PR TITLE
Implemented recently viewed artifacts feature

### LINKED ISSUE #4: [Feature] Implement recently viewed artifacts
To create a new screen which will show artifacts that have been most recently viewed

### PR DESCRIPTION
**Pull request by amlannandy**
**Changes**

Created a new screen which displays the recently viewed artifacts.

**Demo Video**

https://user-images.githubusercontent.com/45410599/104830841-7ffd7880-58a8-11eb-932e-c9e3a5f1f1d6.mp4

closes #4

### CODE REVIEW

**Comment by NishealJ** [comment]
Thanks @amlannandy I'll test this out.

**Comment by NishealJ** [comment]
@amlannandy this PR has changed pubspec.lock is it intentional?

File: pubspec.lock [hunk 1]
```diff
@@ -589,5 +589,5 @@ packages:
     source: hosted
     version: "2.2.1"
 sdks:
-  dart: ">=2.12.0-0.0 <3.0.0"
+  dart: ">=2.10.0-110 <2.11.0"
```
Comment (NishealJ) [inline pubspec.lock:592]: Changes in dart sdk?

**Comment by amlannandy** [comment]
> @amlannandy this PR has changed pubspec.lock is it intentional?

@NishealJ Not intentional. I couldn't run the app before using flutter pub get and this got modified in the mean time it looks like. Should I remove the changes?

**Comment by NishealJ** [comment]
Works fine! Thankyou @amlannandy

**Pull request merged**